plotnine/geoms/annotate.py

```diff
@@ -77,11 +78,22 @@ def __init__(self, geom, x=None, y=None,
                 break
 
         data = pd.DataFrame(position)
-        geom = Registry['geom_{}'.format(geom)]
+        try:
+            if isinstance(geom_cls_or_name, str):
+                geom_to_use = Registry['geom_{}'.format(geom_cls_or_name)]
+            elif isinstance(geom_cls_or_name, type) and issubclass(geom_cls_or_name, geom.geom):
+                geom_to_use = geom_cls_or_name
+            else:
+                raise PlotnineError() # error message comes below
+        except PlotnineError:
+            raise PlotnineError("geom_cls_or_name must either be a geom.geom() " +
+                             "descendant (e.g. plotnine.geom_point), or " +
+                             "a string naming a geom (e.g. 'point', 'text', " +
+                             "...). Was {}".format(repr(geom_cls_or_name)))
```

Comment:
No need for the `+`. And I think the link checker will scream about the indentation.